### PR TITLE
Build tweaks

### LINKED ISSUE #1209: typing index file is missing in npm package
Typings are not picked up in external projects cause the referenced `./dist/node/index.d.ts` does not exist in final package:

Screen Shot 2022-01-11 at 14 04 24

### PR DESCRIPTION
**Pull request by kremalicious**
After tinkering a bit with different options, I was able to just run with the default provided values .

- output browser target instead of Node.js target, ref #1210
- move artifacts folder under /src, so we can ditch specifying the root directory in `tsconfig`, which always comes with more consequences
- remove `os` usage from `ConfigHelper`, ref #1210
- point `typings` to existing typings index file, closes #1209
- build tool tweaks:
  - replicate micro bundle config from https://github.com/developit/microbundle#specifying-filenames-in-packagejson
  - run with suggestions for tsconfig `module` & `target`
  - not sure if `esm` instead of `es` as output is best choice or if there is a difference within `microbundle`
- test tool tweaks:
  - compile test files individually with custom `tsconfig`
  - single `tsconfig` & `mocharc` for both test runs
  - new `npm run mocha` command so we can set some defaults, making it pick up individual `tsconfig`

With all that, these are the only remaining warnings I get during the build (so no warnings about Node.js module usage so remaining `fs` usage might be not problematic at all) and the final file sizes kinda become tolerable:

Screen Shot 2022-01-11 at 16 43 14

### CODE REVIEW

**Comment by kremalicious** [comment]
oh boy, guess now we are own this rabbit hole of figuring out how to transpire only the test files 🥃

File: test/integration/tsconfig.json [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
-      "allowSyntheticDefaultImports": true
-    }
-  }
\ No newline at end of file
+  "compilerOptions": {
+    "resolveJsonModule": true,
+    "lib": ["es6", "es7"],
+    "module": "CommonJS",
+    "target": "ES5",
+    "noUnusedLocals": true,
```
Comment (bogdanfazakas) [inline test/integration/tsconfig.json:7]: you were a bit strict here 😄  it now return errors on issues that are most likely warning for eg:
Error: test/integration/ComputeFlow.test.ts(1,1): error TS6133: 'config' is declared but its value is never read.
but we can handle those and delete unused vars

Comment (kremalicious) [reply test/integration/tsconfig.json]: yeah, figured there's no need for all those individual configs and merged them all into one place in fdcf7df. And made `"noUnusedLocals": false` so I hope now everything gets green again.

**Review by bogdanfazakas — APPROVED** [review]
LGTM nice tweaks 🚀  and if we are fine having artifacts in src folder we can merge this 👍

**Comment by kremalicious** [comment]
OMG, it's green

giphy

Now, final question would be what package market will pick up but since `v4` over there uses webpack 5 in next.js, I am 95% confident it will pick up the CommonJS module referenced under `lib` in package.json

**Pull request merged**